rgd/geodata/management/commands/ingest_s3.py

```diff
@@ -0,0 +1,81 @@
+import multiprocessing
+import re
+from typing import Generator, Optional
+
+import boto3
+import botocore
+import djclick as click
+
+from . import _data_helper as helper
+
+
+def _iter_matching_objects(
+    s3_client, bucket: str, prefix: str, include_regex: str
+) -> Generator[dict, None, None]:
+    paginator = s3_client.get_paginator('list_objects')
+    page_iter = paginator.paginate(Bucket=bucket, Prefix=prefix)
+    include_pattern = re.compile(include_regex)
+    # To avoid `$folder$`, etc. objects
+    exclude_pattern = re.compile(r'\$.+\$')
```

Comment:
This key name pattern is idiosyncratic to the GCS landsat dataset -- is this script designed to be a general purpose S3 ingest script, or specific to that landsat data? If it's meant to be general, let's make the exclude regex another parameter that can be passed to the command.

Reply:
This is intended to be a general-purpose S3 ingest script, so fair enough... I'll compromise and set `\$.+\$` as the default exclude pattern for now. Folks will forget to include it otherwise and it'll result in errors

Reply:
That seems more like a matter for documentation; having a default behavior of excluding any key with at least two non-consecutive `$` characters in it seems to violate the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) IMO. If we really want an easier-to-use special case for the landsat dataset specifically, I believe that should be a separate command whose semantics correspond to the domain model of the landsat dataset, rather than that of S3. Curious what others think, maybe @brianhelba

Reply:
I could change the regex to `\$\w+\$` such that it isn't any two non-consecutive `$` but any word surrounded by `$` to reduce the astonishment factor

Reply:
Or we could leave the exclude regex totally optionally and hardcode to avoid `$folder$` with `.endswith('$folder$')`. I cannot find any documentation for these folder objects on the GCS landsat data

Reply:
I see that `--include-regex` is already a CLI option. Could the same be done for `--include-regex`? This ought to facilitate the capability while avoiding hardcoding anything that's idiosyncratic to one data set.

Reply:
I see the concern about end users forgetting to include it. I think documentation, particularly working examples, would be a good solution. The category of user which we're trying to defensively engineer around is likely to copy-paste any commands from the docs, rather than building up all the arguments thoughtfully.

As long as you give those users the correct examples (including `--exclude-regex`) to copy-paste, I suspect they'll be able to do so without altering anything.

Reply:
Fair enough. I will add some documentation of examples and leave `--exclude-regex` without any default option